src/code/z_kaleido_setup.c

```diff
@@ -89,39 +93,52 @@ void KaleidoSetup_Update(PlayState* play) {
         if (msgCtx && msgCtx) {}
     }
 
-    if ((pauseCtx->state == PAUSE_STATE_OFF) && (pauseCtx->debugEditor == DEBUG_EDITOR_NONE) &&
-        (play->gameOverCtx.state == GAMEOVER_INACTIVE)) {
-        if ((play->transitionTrigger == TRANS_TRIGGER_OFF) && (play->transitionMode == TRANS_MODE_OFF)) {
-            if ((gSaveContext.save.cutsceneIndex < 0xFFF0) && (gSaveContext.nextCutsceneIndex < 0xFFF0)) {
-                if (!Play_InCsMode(play) || ((msgCtx->msgMode != MSGMODE_NONE) && (msgCtx->currentTextId == 0xFF))) {
-                    if ((play->unk_1887C < 2) && (gSaveContext.magicState != MAGIC_STATE_STEP_CAPACITY) &&
-                        (gSaveContext.magicState != MAGIC_STATE_FILL)) {
-                        if (!CHECK_EVENTINF(EVENTINF_17) && !(player->stateFlags1 & PLAYER_STATE1_20)) {
-                            if (!(play->actorCtx.flags & ACTORCTX_FLAG_TELESCOPE_ON) &&
-                                !(play->actorCtx.flags & ACTORCTX_FLAG_PICTO_BOX_ON)) {
-                                if (!play->actorCtx.isOverrideInputOn &&
-                                    CHECK_BTN_ALL(input->press.button, BTN_START)) {
-                                    gSaveContext.prevHudVisibility = gSaveContext.hudVisibility;
-                                    pauseCtx->itemDescriptionOn = false;
-                                    pauseCtx->state = PAUSE_STATE_OPENING_0;
-                                    func_800F4A10(play);
-                                    // Set next page mode to scroll left
-                                    pauseCtx->nextPageMode = pauseCtx->pageIndex * 2 + 1;
-                                    Audio_SetPauseState(true);
-                                }
-
-                                if (pauseCtx->state == PAUSE_STATE_OPENING_0) {
-                                    GameState_SetFramerateDivisor(&play->state, 2);
-                                    if (ShrinkWindow_Letterbox_GetSizeTarget() != 0) {
-                                        ShrinkWindow_Letterbox_SetSizeTarget(0);
-                                    }
-                                    Audio_PlaySfx_PauseMenuOpenOrClose(SFX_PAUSE_MENU_OPEN);
-                                }
-                            }
-                        }
-                    }
-                }
+    if (IS_PAUSED(pauseCtx) || (play->gameOverCtx.state != GAMEOVER_INACTIVE)) {
+        return;
+    }
+
+    if ((play->transitionTrigger != TRANS_TRIGGER_OFF) || (play->transitionMode != TRANS_MODE_OFF)) {
+        return;
+    }
+
+    if ((gSaveContext.save.cutsceneIndex >= 0xFFF0) || (gSaveContext.nextCutsceneIndex >= 0xFFF0)) {
+        return;
+    }
+
+    if (!Play_InCsMode(play) || ((msgCtx->msgMode != MSGMODE_NONE) && (msgCtx->currentTextId == 0xFF))) {
+        if (play->unk_1887C >= 2) {
```

Comment:
Any particular reason this doesn't have an early return like the rest of the function?

Reply:
Because sadly it doesn't match
https://decomp.me/scratch/ZCuoF